esrally/utils/io.py

```diff
@@ -48,22 +50,56 @@ def unzip(zip_name, target_directory):
 
     * zip: Relies that the 'unzip' tool is available on the path
     * bz2: Can be uncompressed using standard library facilities, so no external tool is required.
+    * gz: Can be uncompressed using standard library facilities, so no external tool is required.
+    * tar: Can be uncompressed using standard library facilities, so no external tool is required.
+    * tar.gz Can be uncompressed using standard library facilities, so no external tool is required.
+    * tgz Can be uncompressed using standard library facilities, so no external tool is required.
+    * tar.bz2 Can be uncompressed using standard library facilities, so no external tool is required.
+
+    Did not implement LZMA because LZMAFile is not thread-safe.
 
     The decompression method is chosen based on the file extension.
 
     :param zip_name: The full path name to the file that should be decompressed.
-    :param target_directory: The directory to which files should be decompressed. May or may not exist prior to calling this function.
+    :param target_directory: The directory to which files should be decompressed. May or may not exist prior to calling
+    this function.
     """
     filename, extension = splitext(zip_name)
     if extension == ".zip":
         if not process.run_subprocess_with_logging("unzip %s -d %s" % (zip_name, target_directory)):
             raise RuntimeError("Could not unzip %s to %s" % (zip_name, target_directory))
     elif extension == ".bz2":
-        # We rather avoid external tools as much as possible to simplify Rally's setup, hence we use the library functions
+        # We rather avoid external tools as much as possible to simplify Rally's setup, hence we use the library
+        # functions
         target_file = os.path.join(target_directory, filename)
         with open(target_file, "wb") as extracted, bz2.BZ2File(zip_name, "rb") as file:
             for data in iter(lambda: file.read(100 * 1024), b''):
                 extracted.write(data)
+    elif extension == ".gz":
+        target_file = os.path.join(target_directory, filename)
```

Comment:
Here we have a fair deal of repetition. I wonder whether it would make sense to create a separate internal function for this. Something along the lines of:

```python
def _do_unzip(target_directory, target_filename, compressed_file):
    target_file = os.path.join(target_directory, target_filename)
    with open(target_file, "wb") as extracted:
        for data in iter(lambda: compressed_file.read(100 * 1024), b''):
            extracted.write(data)
```

and then calling it just for each supported extension, e.g. for ".gz" files: `_do_unzip(target_directory, filename, gzip.GzipFile(zip_name, "rb"))`. What do you think?